src/promg/cypher_queries/semantic_header_ql.py

```diff
@@ -319,7 +319,32 @@ def get_reset_created_relation_query(relation_constructor: RelationConstructor)
                      })
 
     @staticmethod
-    def get_create_directly_follows_query(entity: Union[ConstructedNodes, ConstructedRelation], event_label) -> Query:
+    def get_add_duration_query_str(add_duration) -> str:
+        if add_duration:
+            # check the type of the timestamp attributes.
+            # DATE, DATETIME, TIME --> create Duration between first and second
+            # INT, FLOAT --> save difference between second and first
+            add_duration_str = '''
+                , CASE 
+                    WHEN apoc.meta.cypher.type(first.timestamp) IN ["DATE_TIME", "TIME", "DATE"] THEN duration.between(
```

Comment:
Assumption here seems to be that the type of the second timestamp is the same as the type of the first timestamp. Is this guaranteed?

Reply:
The types of timestamps can indeed differ if the events come from datasets. So I added a check for the type of the second timestamp as well. See ff80576.
We'll have to think about how to deal when the timestamp types are different, this should probably be catched when we implement the integrity constraints.